src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,184 @@
+        isSelf(expr.self) &&
+        selfMethodSendFunctions.includes(idText(expr.method)))
+    );
+  }
+
+  private analyzeStatement(
+    statement: AstStatement,
+    warnings: MistiTactWarning[],
+    inLoop: boolean,
+    processedLoopIds: Set<number>,
+  ): void {
+    // check If the statement is a loop!
+    if (this.isLoopStatement(statement)) {
+      // Avoid processing the same loop multiple times
+      if (!processedLoopIds.has(statement.id)) {
+        processedLoopIds.add(statement.id);
+        inLoop = true;
+      }
+    }
+
+    // If stmt is an expr statement we are inside loop
+    if (statement.kind === "statement_expression" && inLoop) {
+      this.checkForSendFunctions(statement.expression, warnings);
+    }
+
+    // Recurse into child stmt
+    const childStatements = this.getChildStatements(statement);
+    for (const child of childStatements) {
+      this.analyzeStatement(child, warnings, inLoop, processedLoopIds);
+    }
+  }
+
+  private checkForSendFunctions(
+    expr: AstExpression,
+    warnings: MistiTactWarning[],
+  ): void {
+    if (this.isSendCall(expr)) {
+      warnings.push(
+        this.makeWarning("Send function called inside a loop", expr.loc, {
+          suggestion:
+            "Consider refactoring to avoid calling send functions inside loops.",
+        }),
+      );
+    }
+    this.checkForSendFunctionsInExpression(expr, warnings);
+  }
+
+  private checkForSendFunctionsInExpression(
+    expr: AstExpression,
+    warnings: MistiTactWarning[],
+  ): void {
+    switch (expr.kind) {
+      case "static_call":
+      case "method_call":
+        if (expr.args && Array.isArray(expr.args)) {
+          for (const arg of expr.args) {
+            this.checkForSendFunctions(arg, warnings);
+          }
+        }
+        break;
+      case "op_binary":
+        this.checkForSendFunctions(expr.left, warnings);
+        this.checkForSendFunctions(expr.right, warnings);
+        break;
+      case "op_unary":
+        this.checkForSendFunctions(expr.operand, warnings);
+        break;
+      case "field_access":
+        this.checkForSendFunctions(expr.aggregate, warnings);
+        break;
+      case "struct_instance":
+        for (const initializer of expr.args) {
+          this.checkForSendFunctions(initializer.initializer, warnings);
+        }
+        break;
+      case "conditional":
+        this.checkForSendFunctions(expr.condition, warnings);
+        this.checkForSendFunctions(expr.thenBranch, warnings);
+        this.checkForSendFunctions(expr.elseBranch, warnings);
+        break;
+      default:
+        break;
+    }
+  }
+
+  private isLoopStatement(statement: AstStatement): boolean {
+    return (
+      statement.kind === "statement_while" ||
+      statement.kind === "statement_repeat" ||
+      statement.kind === "statement_until" ||
+      statement.kind === "statement_foreach"
+    );
+  }
+
+  private getChildStatements(statement: AstStatement): AstStatement[] {
```

Comment:
Why do we need this? Use one of the available statement iterators and visited set.